### PR TITLE
feat: cert manager integration

### PR DESCRIPTION
**Pull request by Piotr1215**
# Content Description
- fixing links and doc structure
- adding more details about cert manager integration

## Preview Link 

[Cert Manager](https://deploy-preview-391--vcluster-docs-site.netlify.app/docs/vcluster/next/integrations/certmanager/)

## Internal Reference

<!--Add the Github or Linear ticket reference>
Closes

### CODE REVIEW

File: vcluster/_fragments/integrations/cert-manager.mdx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+- Ensure that cert-manager is properly installed and running in the host cluster.
+- Check the cert-manager logs in the host cluster for any error messages.
+- Verify that the Issuer and Certificate resources are correctly configured.
+- Ensure that the virtual cluster has the necessary permissions to create and manage certificates.
+
+For more detailed troubleshooting, refer to the [cert-manager troubleshooting guide](https://cert-manager.io/docs/troubleshooting/).
+
+## Config reference
+
+<CertManagerPartial />
```
Comment (hidalgopl) [inline vcluster/_fragments/integrations/cert-manager.mdx:197]: this partial in the preview is empty. Is this intended?

**Review by hidalgopl — APPROVED** [review]

File: vcluster/_fragments/integrations/cert-manager.mdx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  certManager:
+    enabled: true
+```
+
+This configuration enables the integration, imports cluster-scoped ClusterIssuers from the host cluster into the virtual cluster, and exports namespaced Issuers and Certificates from the virtual cluster into the host cluster.
+
+<Flow id="cert-manager-integration">
+ <Step>
+
+  Create the Issuer
```
Comment (deniseschannon) [inline vcluster/_fragments/integrations/cert-manager.mdx:41]: This seems to be a ClusterIssuer that's being created on the host cluster.

1. Can we describe that it's a ClusterIsssuer (the actual CR that's being created)?
2. Can we describe when we apply it, it'll be in the host cluster?

File: vcluster/_fragments/integrations/cert-manager.mdx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+:::note
+This creates a corresponding Issuer in the host cluster.
+:::
+
+</Step>
+<Step>
+
+Create and apply Issuer and Certificate
+
+After the virtual cluster is running, create an Issuer and Certificate inside the virtual cluster. This guide uses a `letsencrypt-staging` issuer for demonstration purposes.
```
Comment (deniseschannon) [inline vcluster/_fragments/integrations/cert-manager.mdx:85]: There's only an example of applying the cert, but not an issuer in the virtual cluster.

Either we ned to update what we're going to do or give an example of how to create the issuer.

File: vcluster/_fragments/integrations/cert-manager.mdx [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+</Step>
+</Flow>
+
+## Verify the integration
+
+<Flow id="verify-integration">
+  <Step>
+Check the status of the Issuer:
+
+   ```bash title="Check Issuer status"
```
Comment (deniseschannon) [inline vcluster/_fragments/integrations/cert-manager.mdx:133]: Where are we checking this? In the host or virtual cluster?

File: vcluster/_fragments/integrations/cert-manager.mdx (hunk 1 above)
Comment (deniseschannon) [inline vcluster/_fragments/integrations/cert-manager.mdx:197]: Screenshot 2024-12-16 at 1 18 30 PM

What's the point of this?

File: vcluster/_partials/config/integrations/certManager.mdx [hunk 5]
```diff
@@ -0,0 +1 @@
+## Cert manager
```
Comment (deniseschannon) [inline vcluster/_partials/config/integrations/certManager.mdx:1]: What is the point of this?

File: vcluster/_fragments/integrations/cert-manager.mdx (hunk 1 above)
Comment (Piotr1215) [reply vcluster/_fragments/integrations/cert-manager.mdx:197]: This is a placeholder for the config will be generated. It doesn't exist yet.

File: vcluster/_fragments/integrations/cert-manager.mdx (hunk 4 above)
Comment (Piotr1215) [reply vcluster/_fragments/integrations/cert-manager.mdx]: I've added clarifications on which cluster we check.

**Review by deniseschannon — APPROVED** [review]

**Pull request merged**